### PR TITLE
Improve support for Leon3 UART device

### PR DESCRIPTION
**Pull request by hikaru1984**
Modify default interrupt vector number from 3 to 2 and add definitions to support multiple UART devices.

### CODE REVIEW

File: .deps/elf.Po [hunk 1]
```diff
@@ -0,0 +1,209 @@
+elf.o: elf.c /usr/include/stdc-predef.h /usr/include/stdio.h \
```
Comment (Lurkerpas) [inline .deps/elf.Po:1]: Should the .deps folder be commited to the repository?

File: .gitignore [hunk 2]
```diff
@@ -50,3 +50,6 @@ modules.order
 Module.symvers
 Mkfile.old
 dkms.conf
+
```
Comment (Lurkerpas) [inline .gitignore:53]: I would also add .deps... or Po

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+SOURCES = elf.c  erc32.c  exec.c  func.c  gr740.c \
+		  greth.c  grlib.c  help.c  interf.c  leon2.c \
+		  leon3.c  linenoise.c  remote.c  riscv.c \
+		  rv32.c sis.c  sparc.c  tap.c
+HEADERS = config.h  elf.h  grlib.h  linenoise.h \
+		  riscv.h  rv32dtb.h  sis.h  sparc.h
+OBJECTS := $(patsubst %.c,%.o, $(SOURCES))
+DEPDIR = .deps
+
+include ./$(DEPDIR)/elf.Po
```
Comment (Lurkerpas) [inline Makefile:16]: Why is this needed?

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,47 @@
+CC := /usr/bin/gcc
+CFLAGS := -O3
+INCL = -I.
+LDFLAGS = -lm -lreadline
+CONFIG = -DHAVE_CONFIG_H 
+DEFS = -DFAST_UART
+SOURCES = elf.c  erc32.c  exec.c  func.c  gr740.c \
```
Comment (Lurkerpas) [inline Makefile:7]: When we have many files and they don't all fit in a single line, it looks nicer if there is one file per line. It is also easier to modify and the diffs look nicer

File: Makefile [hunk 5]
```diff
@@ -0,0 +1,47 @@
+CC := /usr/bin/gcc
```
Comment (Lurkerpas) [inline Makefile:1]: why absolute path?

File: Makefile [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+include ./$(DEPDIR)/sis.Po
+include ./$(DEPDIR)/sparc.Po
+include ./$(DEPDIR)/tap.Po
+
+%.o: %.c
+	$(CC) $(CONFIG) $(DEFS) $(INCL) $(CFLAGS) -MT $@ -MD -MP -MF $(patsubst %.o,$(DEPDIR)/%.Tpo,$@) -c -o $@ $<
+	mv -f $(patsubst %.o,$(DEPDIR)/%.Tpo,$@) $(patsubst %.o,$(DEPDIR)/%.Po,$@)
+
+all: $(OBJECTS)
+	$(CC) $(DEFS) $(CFLAGS) -o sis $(OBJECTS) $(LDFLAGS)
```
Comment (Lurkerpas) [inline Makefile:40]: It would be nice to have BINARY_NAME or PROJECT_NAME or just PROJECT variable that holds the final output name

File: Makefile [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+include ./$(DEPDIR)/linenoise.Po
+include ./$(DEPDIR)/remote.Po
+include ./$(DEPDIR)/riscv.Po
+include ./$(DEPDIR)/rv32.Po
+include ./$(DEPDIR)/sis.Po
+include ./$(DEPDIR)/sparc.Po
+include ./$(DEPDIR)/tap.Po
+
+%.o: %.c
+	$(CC) $(CONFIG) $(DEFS) $(INCL) $(CFLAGS) -MT $@ -MD -MP -MF $(patsubst %.o,$(DEPDIR)/%.Tpo,$@) -c -o $@ $<
```
Comment (Lurkerpas) [inline Makefile:36]: Are our dependencies so complex they need explicit tracing?

File: .gitignore (hunk 2 above)
Comment (hikaru1984) [reply .gitignore:53]: ok

File: Makefile (hunk 3 above)
Comment (hikaru1984) [reply Makefile:16]: it's not - fixed

File: Makefile (hunk 4 above)
Comment (hikaru1984) [reply Makefile:7]: ok

File: Makefile (hunk 6 above)
Comment (hikaru1984) [reply Makefile:40]: fixed

File: .deps/elf.Po (hunk 1 above)
Comment (hikaru1984) [reply .deps/elf.Po:1]: not really - fixed

File: Makefile (hunk 5 above)
Comment (hikaru1984) [reply Makefile:1]: fixed

File: Makefile (hunk 7 above)
Comment (hikaru1984) [reply Makefile]: fixed

**Review by Lurkerpas — APPROVED** [review]
